freqtrade/strategy/interface.py

```diff
@@ -584,6 +584,34 @@ def custom_exit(
         """
         return self.custom_sell(pair, trade, current_time, current_rate, current_profit, **kwargs)
 
+    def custom_orders(
```

Comment:
I'm currently thinking about an elegant way to call 'custom_orders' callback from 'adjust_trade_position'

**Option 1**
`adjust trade position return (None, entry_custom_orders_tag)`
This is the current method, but as the stake will be managed within custom_orders it seem cumbersome

**Option 2**
`adjust trade position return [PositionSizeAction]`

![image](https://github.com/user-attachments/assets/4ac4164f-6f27-477f-b0d6-bb81eaf12d30)

![image](https://github.com/user-attachments/assets/53c9983a-ee35-4e9c-b5e6-123f9bcac23f)


This would be a new way to adjust the trade position, giving more possibilities with clarity.
In this case we could skip 'custom_orders' call as we directly prepare ordersToValidate within the adjust trade position callback.
Alse, we could add a different return type to keep compatibility with the legacy way to adjust position

Reply:
i think your "direction" is wrong (who's calling who).
if you implement `custom_orders()` in your strategy - we'll most likely skip a few callbacks (not sure which ones exactly - but `adjut_trade_position()` is the first that comes to mind).

if you decide to take control of your orders - do it fully (will need good documentation, obviously).
I don't think a mixed mode is supportable / understandable / maintainable (freqtrade places a few orders, the user places other orders, ... ).

there's a few other orders that come to mind as potential candidates (but i didn't fully think that through).

---

As i've said on discord - my intend with this PR is to not really look at it in detail and let you run for the most part - to see what you come up with - before deciding if it's a good approach / something we want to add and support going forward.
This is mostly based on the fact that it started as an empty shell and is moving - so doing a too early "do it this way" or "i don't like this particular part" might influence the outcome.

(I'm not a huge fan of the approach in general at the moment - but good implementations / indeas can convince me eventually).

Reply:
Alright, I did not understand that you were open to such a different and radical approach.
I'm good with it, as it makes things clearer.

My goal is to add a great and maintainable features in the most efficient way possible. Keeping that in mind, I would prefer to have the idea or general approach validated with you before going too deep in any direction.

**Proposal**:

Create a config option: "custom_orders": true

Then, if this is enabled, the following callbacks will be deactivated:

custom_entry_price() and custom_exit_price()
adjust_trade_position()


The "custom_orders" callback will be called at entry and exit. It will also be called at each process loop.
general idea is that if there is a delta between existing exchange orders and what being sent, the delta orders are created. Obviously this should be managed carefully as we don't want to create duplicates. We also need to make sure we stay within the defined stake_amount.

There is also the posibility to create an "adjust_custom_orders" callback in that case it is this callback that will be called at each process loop not "custom_orders".

Reply:
Letme be honest - i am not convinced freqtrade needs this "feature" - independent of "how" it's done.
I am pretty certain that it'll lead to tons of bugs and very difficult to reproduce oddities - as freqtrade no longer controls the flow of orders (each exchange handles things differently - and the validations you write here must cover all of these).

Freqtrade is not a grid trading bot - and to be quite frank, this will quite certainly have a **very** long testing phase - independent of any "tests pass" - with probably tens of iterations until the interface is actually done.

I do have my opinion on this - and as said above - i'm open to have my opinions overturned by a good idea / implementation - but at this point - i don't intend to invest time into it.

---

The approach for this PR is as follows: Do what you think is a must to have working what you think is a great feature - i'll keep myself out of this.
Keep in mind that I'll assume that you'll be available to help users having problems with this (for all supported exchanges, obviously) - so it better have great maintainability and ease of use from a user's perspective :laughing: 

I can still don't like the outcome - as i either don't see the benefit, see it as "bad UX" (it's obscure for the user how it works)  or bad maintainability - or because i don't see myself able (or willing) to support this going forward.

---

If you force me to have an opinion now - then this is not something freqtrade needs - so further discussion is futile and we can drop this idea.

Reply:
I understand that you’re not convinced about the utility of this functionality, but I believe it opens up new possibilities for trading and offers advanced algo traders more control and flexibility.

That said, I recognize that it will introduce more cases to manage and support, and I’ll take responsibility for designing it carefully to address my use case effectively. To minimize risks, I’ll label it as experimental initially, allowing time for testing and refinement.

Once it’s ready, we can revisit whether it’s something you’d consider merging. Thanks for your feedback

Reply:
In this particular case (as i'm not convinced on the need) - it might be better to let your creativity have a go at it.

Otherwise i'll "force" you into a direction - but without actually having a good usecase for it myself necessarily.